src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -333,33 +336,66 @@ private IList<string> GetConfigFilePaths(ISettings settings)
             return settings.GetConfigFilePaths();
         }
 
-        private static PackageReference[] GetPackageReferences(PackageSpec packageSpec)
+        private async Task<IEnumerable<PackageReference>> GetPackageReferences(PackageSpec packageSpec)
         {
             var frameworkSorter = new NuGetFrameworkSorter();
 
+            var lockFilePath = await GetAssetsFilePathAsync();
+            var fileInfo = new FileInfo(lockFilePath);
+            PackageSpec assetsPackageSpec = default;
+            IList<LockFileTarget> targets = default;
+
+            if (fileInfo.Exists && fileInfo.LastWriteTimeUtc > _lastTimeAssetsModified)
+            {
+                var lockFile = new LockFileFormat().Read(lockFilePath);
+                assetsPackageSpec = lockFile.PackageSpec;
```

Comment:
I think in general, you might need to have some sort of a cache that cache the versions from the assets file.

Reply:
So something like, for every tfm, for every package id, have a version.